api/api_ethereum.go

```diff
@@ -1610,14 +1612,18 @@ func EthDoEstimateGas(ctx context.Context, b Backend, args EthTransactionArgs, b
 		args.Gas = (*hexutil.Uint64)(&gas)
 		ret, _, err := EthDoCall(ctx, b, args, rpc.NewBlockNumberOrHashWithNumber(rpc.LatestBlockNumber), nil, 0, gasCap)
 		if err != nil {
+			if errors.Is(err, blockchain.ErrIntrinsicGas) {
+				// Special case, raise gas limit
+				return false, ret, nil, nil
+			}
 			if vm.IsVMError(err) {
 				// If err is vmError, return vmError with returned data
 				return false, ret, err, nil
 			}
 			// Returns error when it is not VM error (less balance or wrong nonce, etc...).
 			return false, nil, nil, err
 		}
-		return true, ret, nil, nil
+		return true, ret, err, nil
```

Comment:
Can this make any difference?

Reply:
No difference. That will be nil because the case(err != nil) already handled before.
If It can be confusing point, I'll drop that change. Thanks.